### PR TITLE
Raise ResponseError unless response is valid

### PR DESCRIPTION
**Pull request by shioyama**
When openlogi returns a request_invalid error it means that the request was invalid, e.g. because the access token was invalid, and in this case we want to raise an error rather than return anything.

For other errors (e.g. data validation errors) we want to return an actual object with errors on it.

### CODE REVIEW

**Comment by degicat** [comment]
@camelmasa Please review this.

File: spec/openlogi/response_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      http_response = double("response", response_body: "\{\"error\":\"invalid_request\"\}")
+      response = Openlogi::Response.new(http_response)
+      expect(response.invalid?).to eq(true)
+    end
+
+    it "returns false if response does not have request_invalid error" do
+      http_response = double("response", response_body: "\{\"error\":\"foo\"\}")
+      response = Openlogi::Response.new(http_response)
+      expect(response.invalid?).to eq(false)
+
```
Comment (camelmasa) [inline spec/openlogi/response_spec.rb]: :scissors:

**Comment by camelmasa** [comment]
LGTM :+1: Please merge it :smiley:

**Comment by shioyama** [comment]
Thanks! I'll merge once specs pass.

**Pull request merged**